src/history.cpp

```diff
@@ -398,7 +398,8 @@ void history_impl_t::save_unless_disabled() {
     if (countdown_to_vacuum < 0) {
         unsigned int seed = static_cast<unsigned int>(time(nullptr));
         // Generate a number in the range [0, kVacuumFrequency).
-        countdown_to_vacuum = rand_r(&seed) / (RAND_MAX / kVacuumFrequency + 1);
```

Comment:
If this doesn't need to be reentrant, it might be worth adding a comment explaining why not.

I struggle with these "marked as obsolete" features; they will never actually be removed, and I think it's reasonable to avoid writing new code with them, but there doesn't seem to be a great deal achieved by excising them from existing codebases.

Reply:
According to the man page, all three functions are reentrant.

From musl:

```
static uint64_t seed;

void srand(unsigned s)
{
        seed = s-1;
}

int rand(void)
{
        seed = 6364136223846793005ULL*seed + 1;
        return seed>>33;
}

static unsigned temper(unsigned x)
{
        x ^= x>>11;
        x ^= x<<7 & 0x9D2C5680;
        x ^= x<<15 & 0xEFC60000;
        x ^= x>>18;
        return x;
}

int rand_r(unsigned *seed)
{
        return temper(*seed = *seed * 1103515245 + 12345)/2;
}
```

I don't see anything that would indicate that rand is not reentrant.

Reply:
rand() is reentrant but this should be replaced with C++ random number generation

Reply:
Yeah, in glib the manpage says "The function rand() is not reentrant, since it uses hidden state that is modified on each call", but STL is much more sensible.